### ISSUE TITLE
Spike: Grouped Radio buttons/Checkboxes

### ISSUE THREAD

**Issue by htmlandbacon** [dwp/design-system#663]
## What

We've seen a couple of examples of large lists of radios being split up by headers (within the radio) we should extract this as a component to test.

Most recently 'Determine a customer's habitual residence' sent an example.

We've also seen a version of this in PIP with grouped checkboxes.

## Why

In agent interfaces with longer lists it is better to have clear sections to help people navigate to what they need to select.

The example given had content issues, but the example represented to the accessibility team seemed to resolve this issue.

The other assumption is this might be better as guided questions.

## Done when

- [ ] Example html 
- [ ] Example component
- [ ] Reviewed and decided next steps

## Outcomes

We should have a clear idea of what to do with this suggestion.

## Anything else

- [GDS Design System - Radios](https://design-system.service.gov.uk/components/radios/)
- [GitHub - Radios](https://github.com/alphagov/govuk-design-system-backlog/issues/59)
